paymentsheet/src/main/java/com/stripe/android/paymentsheet/viewmodels/BaseSheetViewModel.kt

```diff
@@ -157,6 +157,9 @@ internal abstract class BaseSheetViewModel(
     internal val selection: StateFlow<PaymentSelection?> = savedStateHandle
         .getStateFlow<PaymentSelection?>(SAVE_SELECTION, null)
 
+    internal val mostRecentlySelectedSavedPaymentMethod: StateFlow<PaymentMethod?> =
+        savedStateHandle.getStateFlow(SAVED_PM_SELECTION, null)
+
```

Comment:
This is necessary for this scenario:
- the currently selected payment method is second in the `paymentMethods` list
- you select a new payment method

if we didn't separately saved the most recently selected saved payment method, then we would start displaying the first payment method in the `paymentMethods` list

Reply:
Added a note in the end to end tests jira to make sure we test this scenario: https://jira.corp.stripe.com/browse/MOBILESDK-2152